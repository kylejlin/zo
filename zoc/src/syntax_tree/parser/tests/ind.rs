use super::*;

#[test]
fn ind() {
    let src = r#"
(
ind

Set0

"Nat"

// Index types
()

// Variant constructor defs
(
    (() ())
    ((0) ())
)
)"#;
    let tokens = lex(src).unwrap();
    let cst = parse(tokens).unwrap();
    insta::assert_debug_snapshot!(&cst);
}

#[test]
fn ind_nonliteral_universe() {
    let src = r#"
(
ind

1

"Nat"

// Index types
()

// Variant constructor defs
(
    (() ())
    ((0) ())
)
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}

#[test]
fn ind_nonliteral_name() {
    let src = r#"
(
ind

Set0

2

// Index types
()

// Variant constructor defs
(
    (() ())
    ((0) ())
)
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}
