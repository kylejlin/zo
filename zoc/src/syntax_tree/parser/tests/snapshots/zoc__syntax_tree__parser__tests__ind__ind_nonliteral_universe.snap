---
source: zoc/src/syntax_tree/parser/tests/ind.rs
expression: "&err"
---
Some(
    Number(
        NumberLiteral {
            value: 1,
            span: (
                ByteIndex(
                    8,
                ),
                ByteIndex(
                    9,
                ),
            ),
        },
    ),
)
